runtime/vm/ModularityHashTables.c

```diff
@@ -24,31 +24,19 @@
 #include "j9protos.h"
 #include "ut_j9vm.h"
 
-static j9object_t moduleHashGetName(const void* entry);
 static UDATA packageHashFn(void *key, void *userData);
 static UDATA packageHashEqualFn(void *leftKey, void *rightKey, void *userData);
 static UDATA moduleHashFn(void *key, void *userData);
 static UDATA moduleHashEqualFn(void *leftKey, void *rightKey, void *userData);
 static UDATA moduleExtraInfoHashFn(void *key, void *userData);
 static UDATA moduleExtraInfoHashEqualFn(void *tableNode, void *queryNode, void *userData);
 
-static j9object_t
-moduleHashGetName(const void* entry)
-{
-	const J9Module** const modulePtr = (const J9Module**)entry;
-	const J9Module* const  module = *modulePtr;
-	j9object_t moduleName = module->moduleName;
-
-	return moduleName;
-}
-
 static UDATA 
 moduleHashFn(void *key, void *userData) 
 {
-	J9JavaVM* javaVM = (J9JavaVM*) userData;
-	j9object_t name = moduleHashGetName(key);
-
-	return javaVM->memoryManagerFunctions->j9gc_stringHashFn(&name, userData);
+	const J9Module** const modulePtr = (const J9Module**)key;
+	
+	return (UDATA)(*modulePtr);
```

Comment:
Thanks @DanHeidinga , `J9Module` objects are allocated with `pool_newElement(vm->modularityPool)` as following:
https://github.com/eclipse/openj9/blob/d12dbf8365974daaa39257ad035cab94000c2a5c/runtime/vm/jvminit.c#L2196
https://github.com/eclipse/openj9/blob/d12dbf8365974daaa39257ad035cab94000c2a5c/runtime/vm/jvminit.c#L2203
https://github.com/eclipse/openj9/blob/d12dbf8365974daaa39257ad035cab94000c2a5c/runtime/j9vm/java11vmi.c#L252
So there is no stack-allocation of `J9Module`. In addition existing `moduleExtraInfoHashFn` & `moduleExtraInfoHashEqualFn` have performed hash on `J9Module` pointer since modularity era.